<TITLE>Supplement to the MacClade Book</TITLE><a href="macclade.html"><IMG ALIGN=MIDDLE SRC="pictures/macclade.button.gif"></a><a href="macclade.bugs.html"><IMG ALIGN=MIDDLE SRC="pictures/bug.button.gif"></a><HR><h2> Supplement to the second printing of the MacClade Book</h2><p>For MacClade version 3.06<BR>Wayne P. Maddison and David R. Maddison<BR>14 August 1996<BR><p>This Supplement describes aspects of the program not described in the book. It is organized by chapters, with page numbers referring to those in the book.  <p>This is the supplement for the second printing of the MacClade book (in the first printing there were more typographical errors).  First printings of the book will have, as the bottom line on the copyright page, a string of numbers ending in "1", that is, "5 4 3 2 1".  Second printings will have the string "5 4 3 2".  If you have a first printing, please see the <a href="supplement.html">Supplement to the First Printing of the MacClade Book</a>.  <P><p>For technical support (including bug reports) please use the followingelectronic mail address: <p><tt><b><p>clade@arizona.edu</b></tt><p><p>As we do not have staff available to help us with technical support, we wouldappreciate if you contact us by e-mail, not telephone.<p><H3>Contents</H3><UL><LI><a href="#preface">Preface</a><LI><a href="#5">5. Reconstructing Character Evolution Using Parsimony</a><LI><a href="#7">7.  Using MacClade on Your Macintosh</a><LI><a href="#9">9.  Importing and Exporting Text File</a><LI><a href="#10">10. The Data Editor </a><LI><a href="#11">11. Assumptions About Characters</a><LI><a href="#12">12. Trees and Tree Manipulation</a><LI><a href="#13">13. Tracing Character Evolution</a><LI><a href="#14">14. Basic Tree and Character Statistics</a><LI><a href="#15">15. Charting Tree and Character Statistics</a><LI><a href="#16">16. Continuous Characters</a><LI><a href="#18">18. Generating Random Data and Random Trees </a><LI><a href="#19">19. Using MacClade with Molecular Data</a><LI><a href="#20">20. Recording Your Work: Printing, Graphics, and Text Files</a><LI><a href="#appendix">Appendix 1: File Format</a><LI><a href="#references">references</a></UL><H3><a name="5">5. Reconstructing Character Evolution Using Parsimony</a></H3>In the boxes describing algorithms on pages 92 to 96, "{0...maxstate}" appears.In this context, "maxstate" means the maximum observed or allowed state in thecharacter (this is not the same as the MAXSTATE option referred to elsewhere inthe book).<p><H3><a name="7">7.  Using MacClade on Your Macintosh</a></H3>The font "MacClade Small" is now housed in a separate file, called "MacClade Font File".  This file should be placed in the same folder as MacClade or in the System Folder.  For machines running System 7 or later, the file may be placed in the Fonts Folder.  For machines running System 6 or earlier, the fonts may be moved into the System file using the Font/DA Mover, as descri bed in the manual for your Macintosh.<P><H3><a name="9">9.  Importing and Exporting Text File</a></H3>You can now choose to export into a simple table file (p. 156-158) anyfootnotes contained in your matrix, by checking "include footnotes" in the<b>Simple Table</b> dialog box.<p>Molecular sequence data can now be exported to the format used by MEGA version 1 (Kumar et al., 1993), using the <b>MEGA 1</b> item in the <b>Export File </b>submenu in the <b>File </b>menu.  If some terminal taxa are polymorphic or have partial uncertainty at some sites, then MEGA 1 will not be able to read the files without modification.  If MacClade detects polymorphisms or partial uncertainties, it will provide three options: to cancel the export, to convert polymorphisms and uncertainties to missing data, or to write the polymorphisms and uncertainties using IUPAC ambiguity codes.    The latter option will produce files that will not be compatible with MEGA version 1, but which will be compatible with a later version (Kumar, pers. comm.).<H3><a name="10">10. The Data Editor </a></H3>Block Moves</b> (p. 168, 329).  Holding down the Option key removes theconstraint imposed by the edges of the matrix while moving blocks with themouse, but not with the arrow keys.<p><b><p>Compress Characters</b> (in the <b>Utilities</b> menu).  Choosing <b>Compress Characters</b> will cause MacClade tosearch for characters with identical character states; MacClade will thencompress these identical characters into one.  Excluded characters will bere-included and compressed.  The type of the compressed character will be thetype of the first encountered character of that sort; the weight will be thesum of the weights of the compressed characters.  Thus, if there were threecharacters (each given weight 1), all of which had A for taxa 1 and 2 but G forthe remaining taxa, then MacClade would compress these three characters intoone, and give that single compressed character a weight of 3.<b><p><p>Filter Taxa </b>(p. 172).  If you ask to filter taxa (and "consider taxaredundant even if states are not identical..." is not chosen), then MacCladewill report the number of "extra" taxa. For example, if a matrix contains 4identical taxa MacClade will report "Number of redundant taxa: 3". If you dochoose "consider taxa redundant even if states are not identical...", thenMacClade will report the number of redundant pairs, in this case 6.<p><b><p>Pasting Rows and Columns </b>(p. 184)<b>.</b>  If you attempt to paste aclipboard of entire rows or columns containing more rows or columns than thecurrent selection, MacClade will automatically insert extra rows or columns toaccommodate the pasted elements. <p><p>You can now ask MacClade to speak your data matrix.  This command, which was present in test copies of MacClade in 1987-1989, has been resurrected now that Apple has released a new speech synthesizer.  To ask MacClade to speak your matrix, you need to be running System 7.1 or later, and to have the system extensions Macintalk and the Speech Manager installed in the Extensions Folder in your System Folder.  If you do not have copies, you can currently (May 1995) download the Text to Speech software package from Apple Computer's World Wide Web side at the URL "<a href="http://www.info.apple.com">http://www.info.apple.com"</a>.  It may be possible to purchase this software from Apple Computer.  Once you have installed these, you will need to restart your machine to have them take effect.<p><b><p>Speak Matrix</b> in the <b>Utilities</b> menu will then cause MacClade to readyour matrix. If you have a single cell selected in the matrix, MacClade willbegin speaking from that point; otherwise, it will begin at the start of thematrix. Speaking will continue until you hold the mouse button down for awhile, or until you type command-period.Currently, you have very little control over how MacClade speaks the matrix.For example, you cannot choose to use different voices; MacClade uses thedefault voice.  You can only control the timing of pauses in reading thematrix, using the <b>Speech Options</b> dialog box in the <b>Utilities</b>menu. If you want MacClade to pause after every cell, enter "1" in the "every _items" box.<p><H3><a name="12">12. Trees and Tree Manipulation</a></H3>The <b>Include-Exclude Taxa</b> dialog box now has an All button.  Pressingthis button will cause the entire list of excluded taxa to be selected.<p><H3><a name="13">13. Tracing Character Evolution</a></H3><b>Trace All Changes</b> (pp. 254-261).  A number of aspects of this feature havechanged, as described below.<p><p>The design of the <b>Trace Labeling</b> dialog box (pp. 256-258) has changed;it now looks like this:<p><IMG SRC="pictures/trace.all.changes.gif"><p>There is one minor new feature available: if you display the characterschanging along a branch with bars (tick marks), you can choose to have the barsunlabelled.  Also, if you choose to label branches by the amount of change, andthe changes on a particular branch are not calculated (for example, because thebranch is part of a polytomy), then the branch will be labeled "NC" in version3.02 rather than "not calc." as in versions 3.0 and 3.01.  This was done toprevent overlapping labels on the screen.<p><p>You can now request that MacClade draw the length of each branch proportionalto the number of changes reconstructed along the branch. To do this, first makesure <b>Trace All Changes</b> is on, then choose <b>Tree Shape &amp; Size</b>from the Display menu.  This dialog now includes a third shape icon, on theright hand side:<p><IMG SRC="pictures/tree.shape.gif"><p><p>Select it, and press OK; MacClade will then draw the branches with variedbranch lengths.  If minimum-average-maximum changes are calculated, thenMacClade will draw the branch lengths proportional to the maximum changes.(Note that to the length of each branch is added a small  amount [4 pixels].This enables even zero-length branches to be picked up and moved around.)  Whenproportional branch lengths are shown in the tree window, MacClade willautomatically adjust the vertical scale of the branches so that bars and labelson branches will be visible.  However, this is not done if you have askedMacClade to weight  changes by the cost of changes or character weights in the<b>All Changes Options</b> dialog box. <p><b><p>Trace All States</b> (p. 261).  A character with all missing data in theobserved taxa in the current tree will be reconstructed as having any possiblestate at each node in the tree for molecular data (e.g., the MPR set at eachnode will be "A or C or G or T"). Thus these MPR sets will contribute to statesand frequencies listed by Trace All States.<p><H3><a name="15">15. Charting Tree and Character Statistics</a></H3>A text file of chart calculations (p. 275) can now be written for <b>StateChanges &amp; Stasis charts</b> (pp. 287-293) calculated over a single tree ifminimum-maximum changes are requested.<p><p><p>If you use <b>Chart To Weight Set</b> (pp. 283-284), then the characters in anyinterval with 0 observed steps or changes in a chart showing steps or changeswill be assigned a weight equal to the maximum value of the scale chosen. Thus,0 steps using the W = 1/S weight function and a weight scale from 1 to 10 willbe assigned a weight of 10, not a weight of [[infinity]]. Characters with thenext-lowest number of steps or changes will be assigned the same, maximalvalue.<p><p>In the <b>Chart To Weight Set</b> (pp. 283-284) and <b>Chart To Type Set</b>(p. 295), MacClade will beep if you enter a value for the upper end of theweight scale that is too high (see p. 198 for weights, p. 206 for types).  <p><p>If you ask to have the <b>States</b> chart (p. 296) displayed usingpercentages, then in the bar chart and table the values are shown with respect to the pool of observed states (not includingmissing data and gaps). Thus, if frequencies in the data matrix are 20% foreach of A, C, G, T, and gaps, then the charts would show frequencies of A, C,G, and T of 25%. The textual summary calculates percentages in two ways: (1) as in the bar charts and tables; (2)with respect to the total number of data matrix entries (including missing dataand gaps). The latter method would calculate percentages of A, C, G, and T as20% for the example given. <p><p>In the <b>Compare 2 Trees</b> and <b>Compare 2 Tree Files</b> charts (pp.296-297), the vertical axis options dialog box gives you the option to have the bars indicate either which characters areshorter in a tree or tree file (as illustrated on pp. 296-297 of book), orwhich are longer.<p><H3><a name="16">16. Continuous Characters</a></H3>The maximum number of continuous characters has increased from four to ten.These are now chosen in the <b>Edit Continuous</b> and <b>Trace Continuous</b>dialog boxes using scroll bars, not radio buttons.<p><p>Continuous values can be from 0.0 to 9999.9. Larger values and negative valuesare not allowed.<p><p>The <b>Edit Continuous</b> dialog box (p. 303) allows you to indicate a rangefor a single terminal taxon by using "to" or "-", as in "0.13 to 0.25" or "2.31- 3.44". You can also indicate small or large numbers using exponentialnotation, as in "3.5e3" (for 3500.0) or "2.8e-6" (for 0.0000028). However, inthe same entry you cannot use both negative exponents and "-" to indicate arange. Thus "1.4e-6 - 2.8e-6" is not allowed and will not be interpretedcorrectly.<p><p><H3><a name="18">18. Generating Random Data and Random Trees </a></H3><B>Fill data cells with observed states</b>. If you select a block of cells in theeditor, hold down the Option key, and choose <b>Fill With Observed</b> from the<b>Utilities</b> menu, you will be presented with a dialog box. If you pressOK, MacClade will fill the selected cells of each character with random data.If you choose "with probabilities matching observed frequencies", MacClade willfirst tabulate the relative frequencies of each of the states observed in acharacter in all taxa, and fill the selected cells of that character withrandom data, with the probability of filling a cell with a state matching theobserved frequency of each state. For example, if a given nucleotide site had30% A, 10% G, and 60% T, then the selected cells would be filled with thestates A, C, G, and T with probability 0.30, 0.00, 0.10, and 0.60. In contrast,if you choose "with equal probability for each observed state", then theprobabilities of each state will be equal. In the example given, the cellswould be filled with states A, C, G, and T with probabilities 0.33, 0.00, 0.33,and 0.33 (note that the probability of C is 0 as there were no observed C's forthis site). If a character has only missing data or gaps, then the selectedcells will be filled with missing data.<p><b><p>Evolve Characters</b> (pp. 320-322). The number of segments that can beassigned to each branch for the <b>Evolve Characters</b> feature cannot be morethan 1000. Also, the WARNING on p. 321 of the book is no longer applicable, asthe number of segments assigned to each branch is now saved for the currenttree. Thus, the number of segments on each branch should be saved when you togo to the data editor and back, when you save and reopen the file, and so on.However, the numbers of segments are saved only for the tree in the tree window(other trees stored in the file are not stored with their numbers of segments).However, in special cases in which a tree change is Undone, the evolve segmentsmay revert to all 1. Thus, you should still check to see that the segments areas you want them just before using them.  Also, there is now a "save report"check box that allows you to choose whether or not to save a report of thesimulation (versions 3.0-3.02 always saved a report).<p><p><H3><a name="19">19. Using MacClade with Molecular Data</a></H3>For protein-coding nucleotide sequence data, the site number will be displayedin varied colors in the data editor on a color monitor: blue, green, and redfor the first, second, and third codon positions respectively. At the bottomleft of the editor, beside the autotab buttons, the numbers "1 2 3" will appearin these colors to provide a reminder about the correspondence between colorsand codon positions.<p><p>If <b>Translate to Protein</b> (p. 327) is used, MacClade will automaticallyname the resulting protein sites. Sites translated from the firstprotein-coding block in the file will be named 1.1, 1.2, 1.3, and so on. If inthe original nucleotide sequence there was more than one protein-coding region(separated by non-coding regions), then sites in the second translated regionwill be named 2.2, 2.3, 2.4, and so on.<p><p>The <b>Genetic Code</b> dialog box has been altered to allow for several new pre-defined genetic codes.  In versions 3.0-3.04, you selected the two predefined genetic codes (universal and mitochondrial) using buttons in the dialog box; in version 3.05-3.06, these have been replaced by a pop-up menu, with eight codes.  These include universal, universal extended (equivalent to the universal of versions 3.0 to 3.04), Drosophila mitochondrial DNA, Drosophila mitochondrial DNA extended, mammalian mitochondrial DNA, mammalian mitochondrial DNA extended (equivalent to the mtDNA of versions 3.0 to 3.04), and yeast mitochondrial DNA (Osawa et al., 1992).  The extended codes are those in which the disjunct amino acids are split into distinct states (see p. 332 of the book).   <p><p>You cannot ask MacClade to create a protein-parsimony user-defined type (p.333) if the tree window is on the screen.  This can be done only if the dataeditor is on the screen.<p><p>See also the note regarding Trace All States under Chapter 13.<p><H3><a name="20">20. Recording Your Work: Printing, Graphics, and Text Files</a></H3>Of the various options for altering display and printing of trees (pp. 232-233,245-246, 256-258, 343-352), some can be controlled in the tree window, some inthe <b>Print Tree</b> dialog box. The rules are:<p><p>1. 	The controls are separate for the taxon name fonts, whether patterns,colors, or grays are used in tracing branches, and all options in the <b>TraceLabeling</b> dialog box for Trace All Changes except whether ambiguous changesare distinguished. Changing one of these options in the tree window will notaffect values in the <b>Print Tree</b> dialog box.<p><p>2. 	Controls to change other options in the <b>Trace Labeling</b> dialog boxesare present in both windows, and changing values in one window affects thevalues in the other. <p><p>3. 	Controls for showing node numbers, showing evolve segments, adjusting thecontent of the taxon labels, and the patterns or colors used in tracing areavailable only in the tree window, not in the <b>Print Tree</b> dialog box.<p><p>4. 	Some options are relevant to only one of the windows, and only havecontrols in that window (e.g., branch lengths).<p><H3><a name="appendix">Appendix 1: File Format</a></H3>The NEXUS file format has changed slightly in the last year, and these changesare reflected in 3.05 and 3.06.  These changes were made in the course of preparing theformal description of the NEXUS format for publication.  They primarily affectthe genetic code and codon positions, continuous data, user-defined types andweights with real values, italicized taxon names, and character sets.  All ofthe new features of the file format will be enabled if you choose the "useupdated NEXUS elements" in the Nexus Options dialog box.  Some of the newfeatures (genetic code, codons, italicized taxon names) will be used even ifyou do not check that option.  Use of these new features may make your fileunreadable by earlier versions of MacClade.  <p><p>In addition, the rules regarding punctuation have changed.  By default, MacClade 3.06 uses the old punctuation rules on reading a file, the new ones on writing a file (these new files are perfectly legible to MacClade 3.0-3.04).  (The new rules dictate that +, `, <, and > are punctuation, which will break a token; the old rules did not specify these as punctuation.)  If you want to force MacClade to use the new rules when reading a file, you can do this by going to the <b>Save Preferences</b> dialog box in the File menu, uncheck "old punctuation", and press the "Save Preferences" button.  The new rules are not currently relevant for any program.<p><H3><a name="references">References</a></H3>The following are two new references:<P>Kumar, S., K. Tamura, and M. Nei.  1993.  MEGA: Molecular Evolutionary Genetics Analysis, version 1.0. The Pennsylvania State University, University Park, PA 16802.<P>Osawa, S., T.H. Jukes, K. Watanabe, and A. Muto.  1992.  Recent evidence for the evolution of the genetic code.  Microbiol. Rev. 56:229-264<P>