<HTML><!--This file created 27/10/2000 11:29 AM by Claris Home Page version 3.0--><HEAD>   <TITLE>Supplement to the MacClade Book</TITLE>   <META NAME=GENERATOR CONTENT="Claris Home Page 3.0">   <X-CLARIS-WINDOW TOP=66 BOTTOM=870 LEFT=8 RIGHT=538>   <X-CLARIS-TAGVIEW MODE=minimal></HEAD><BODY BGCOLOR="#FFFFFF"><P><A HREF="macclade3.html"><IMG SRC="pictures/macclade.button.gif" WIDTH=23 HEIGHT=23 X-CLARIS-USEIMAGEWIDTH X-CLARIS-USEIMAGEHEIGHT ALIGN=middle></A><HR></P><H2>Supplement to the first printing of the MacClade 3 Book</H2><P>For MacClade version 3.08<BR>Wayne P. Maddison and David R. Maddison<BR>28 April 1999<BR></P><P>Since the MacClade book went to press, there have been a number ofminor changes to the program, and a number of errors detected in thebook. This supplement describes aspects of the program not describedin the book, and corrects known errors in the book. It is organizedby chapters, with page numbers referring to those in the book. Thissupplement applies to the first printing of the book.</P><P>This is the supplement for the first printing of the MacClade book(in the second printing a few of the typographical errors mentionedherein were fixed). First printings of the book will have, as thebottom line on the copyright page, a string of numbers ending in "1",that is, "5 4 3 2 1". Second printings will have the string "5 4 32". If you have a second printing, this information is alreadyincluded as an addendum to the book.&nbsp;</P><H3>Contents</H3><UL>   <LI><A HREF="#preface">Preface</A></LI>      <LI><A HREF="#5">5. Reconstructing Character Evolution Using   Parsimony</A></LI>      <LI><A HREF="#7">7. Using MacClade on Your Macintosh</A></LI>      <LI><A HREF="#9">9. Importing and Exporting Text File</A></LI>      <LI><A HREF="#10">10. The Data Editor </A></LI>      <LI><A HREF="#11">11. Assumptions About Characters</A></LI>      <LI><A HREF="#12">12. Trees and Tree Manipulation</A></LI>      <LI><A HREF="#13">13. Tracing Character Evolution</A></LI>      <LI><A HREF="#14">14. Basic Tree and Character Statistics</A></LI>      <LI><A HREF="#15">15. Charting Tree and Character   Statistics</A></LI>      <LI><A HREF="#16">16. Continuous Characters</A></LI>      <LI><A HREF="#18">18. Generating Random Data and Random Trees   </A></LI>      <LI><A HREF="#19">19. Using MacClade with Molecular Data</A></LI>      <LI><A HREF="#20">20. Recording Your Work: Printing, Graphics, and   Text Files</A></LI>      <LI><A HREF="#appendix">Appendix 1: File Format</A></LI>      <LI><A HREF="#references">references</A></LI></UL><H3><A NAME=preface></A>Preface</H3><P>One of the most important acknowledgements was unfortunatelyomitted in the Preface, namely that to Dan Fisher, for being willingto publish an outline of his ideas on stratocladistics in theMacClade book (Chapter 6), and for supporting our incorporation ofthe stratigraphic character type into MacClade.</P><P>We would like to thank Jeremy Ahouse, Darrel Frost, Ray Phillips,and David Stern for aiding us in fixing some bugs in MacClade.</P><P>&nbsp;</P><H3><A NAME=5></A>5. Reconstructing Character Evolution UsingParsimony</H3><P>On page 81, the figure in the margin showing the pathways betweenstates has states 1 and 2 reversed.</P><P>&nbsp;</P><P>In the boxes describing algorithms on pages 92 to 96,"{0...maxstate}" appears. In this context, "maxstate" means themaximum observed or allowed state in the character (this is not thesame as the MAXSTATE option referred to elsewhere in the book).</P><P>&nbsp;</P><P>On page 97, the three "+" symbols in the box for stratigraphiccharacters should be union symbols ("&#91;&#91;union&#93;&#93;").</P><P>&nbsp;</P><P>On page 102, the headings for the final two columns under "Finalpass assignments to N" are reversed. That is, the second columnshould have the heading "FM={1}", and the third column should havethe heading "FM={0,1}".</P><P>&nbsp;</P><H3><A NAME=7></A>7. Using MacClade on Your Macintosh</H3><P>In the figure "Limits on matrix size in MacClade" (p. 135), thelegend at the upper right has the symbols for 2Mb and 4Mb inverted.Thus, 2Mb is represented in the chart by black squares, and 4Mb byopen triangles.</P><P>&nbsp;</P><H3><A NAME=9></A>9. Importing and Exporting Text File</H3><P>You can now choose to export into a simple table file (p. 156-158)any footnotes contained in your matrix, by checking "includefootnotes" in the <B>Simple Table</B> dialog box.</P><P>Molecular sequence data can now be exported to the format used byMEGA version 1 (Kumar et al., 1993), using the <B>MEGA 1</B> item inthe <B>Export File </B>submenu in the <B>File </B>menu. If someterminal taxa are polymorphic or have partial uncertainty at somesites, then MEGA 1 will not be able to read the files withoutmodification. If MacClade detects polymorphisms or partialuncertainties, it will provide three options: to cancel the export,to convert polymorphisms and uncertainties to missing data, or towrite the polymorphisms and uncertainties using IUPAC ambiguitycodes. The latter option will produce files that will not becompatible with MEGA version 1, but which will be compatible with alater version (Kumar, pers. comm.).</P><H3><A NAME=10></A>10. The Data Editor</H3><P>Block Moves (p. 168, 329). Holding down the Option key removes theconstraint imposed by the edges of the matrix while moving blockswith the mouse, but not with the arrow keys.</P><P>&nbsp;</P><P><B>Compress Characters</B> (in the <B>Utilities</B> menu).Choosing <B>Compress Characters</B> will cause MacClade to search forcharacters with identical character states; MacClade will thencompress these identical characters into one. Excluded characterswill be re-included and compressed. The type of the compressedcharacter will be the type of the first encountered character of thatsort; the weight will be the sum of the weights of the compressedcharacters. Thus, if there were three characters (each given weight1), all of which had A for taxa 1 and 2 but G for the remaining taxa,then MacClade would compress these three characters into one, andgive that single compressed character a weight of 3.</P><P>&nbsp;</P><P><B>Filter Taxa </B>(p. 172). If you ask to filter taxa (and"consider taxa redundant even if states are not identical..." is notchosen), then MacClade will report the number of "extra" taxa. Forexample, if a matrix contains 4 identical taxa MacClade will report"Number of redundant taxa: 3". If you do choose "consider taxaredundant even if states are not identical...", then MacClade willreport the number of redundant pairs, in this case 6.</P><P>&nbsp;</P><P><B>Pasting Rows and Columns </B>(p. 184)<B>.</B> If you attempt topaste a clipboard of entire rows or columns containing more rows orcolumns than the current selection, MacClade will automaticallyinsert extra rows or columns to accommodate the pasted elements.</P><P>&nbsp;</P><P>You can now ask MacClade to speak your data matrix. This command,which was present in test copies of MacClade in 1987-1989, has beenresurrected now that Apple has released a new speech synthesizer. Toask MacClade to speak your matrix, you need to be running System 7.1or later, and to have the system extensions Macintalk and the SpeechManager installed in the Extensions Folder in your System Folder. Ifyou do not have copies, you can currently (May 1995) download theText to Speech software package from Apple Computer's World Wide Webside at the URL "<A HREF="http://www.info.apple.com">http://www.info.apple.com"</A>.It may be possible to purchase this software from Apple Computer.Once you have installed these, you will need to restart your machineto have them take effect.</P><P>&nbsp;</P><P><B>Speak Matrix</B> in the <B>Utilities</B> menu will then causeMacClade to read your matrix. If you have a single cell selected inthe matrix, MacClade will begin speaking from that point; otherwise,it will begin at the start of the matrix. Speaking will continueuntil you hold the mouse button down for a while, or until you typecommand-period. Currently, you have very little control over howMacClade speaks the matrix. For example, you cannot choose to usedifferent voices; MacClade uses the default voice. You can onlycontrol the timing of pauses in reading the matrix, using the<B>Speech Options</B> dialog box in the <B>Utilities</B> menu. If youwant MacClade to pause after every cell, enter "1" in the "every _items" box.</P><P>&nbsp;</P><H3><A NAME=11></A>11. Assumptions About Characters</H3><P>The <B>Select</B> submenu (p. 196) now has an <B>All Invariant</B>item added to it. If this is chosen, then all characters containingat most one state will be selected in the character status window.Characters with missing data or gaps in addition to a single statewill be included, as will characters with only missing data orgaps.</P><P>&nbsp;</P><H3><A NAME=12></A>12. Trees and Tree Manipulation</H3><P>The <B>Include-Exclude Taxa</B> dialog box now has an All button.Pressing this button will cause the entire list of excluded taxa tobe selected.</P><P>&nbsp;</P><H3><A NAME=13></A>13. Tracing Character Evolution</H3><P><B>Trace All Changes</B> (pp. 254-261). A number of aspects ofthis feature have changed, as described below.</P><P>&nbsp;</P><P>The design of the <B>Trace Labeling</B> dialog box (pp. 256-258)has changed; it now looks like this:</P><P><IMG SRC="pictures/trace.all.changes.gif" WIDTH=503 HEIGHT=234 X-CLARIS-USEIMAGEWIDTH X-CLARIS-USEIMAGEHEIGHT ALIGN=bottom></P><P>There is one minor new feature available: if you display thecharacters changing along a branch with bars (tick marks), you canchoose to have the bars unlabelled. Also, if you choose to labelbranches by the amount of change, and the changes on a particularbranch are not calculated (for example, because the branch is part ofa polytomy), then the branch will be labeled "NC" in version 3.02rather than "not calc." as in versions 3.0 and 3.01. This was done toprevent overlapping labels on the screen.</P><P>&nbsp;</P><P>You can now request that MacClade draw the length of each branchproportional to the number of changes reconstructed along the branch.To do this, first make sure <B>Trace All Changes</B> is on, thenchoose <B>Tree Shape &amp; Size</B> from the Display menu. Thisdialog now includes a third shape icon, on the right hand side:</P><P><IMG SRC="pictures/tree.shape.gif" WIDTH=343 HEIGHT=211 X-CLARIS-USEIMAGEWIDTH X-CLARIS-USEIMAGEHEIGHT ALIGN=bottom></P><P>&nbsp;</P><P>Select it, and press OK; MacClade will then draw the branches withvaried branch lengths. If minimum-average-maximum changes arecalculated, then MacClade will draw the branch lengths proportionalto the maximum changes. (Note that to the length of each branch isadded a small amount &#91;4 pixels&#93;. This enables evenzero-length branches to be picked up and moved around.) Whenproportional branch lengths are shown in the tree window, MacCladewill automatically adjust the vertical scale of the branches so thatbars and labels on branches will be visible. However, this is notdone if you have asked MacClade to weight changes by the cost ofchanges or character weights in the <B>All Changes Options</B> dialogbox.</P><P>&nbsp;</P><P><B>Trace All States</B> (p. 261). A character with all missingdata in the observed taxa in the current tree will be reconstructedas having any possible state at each node in the tree for moleculardata (e.g., the MPR set at each node will be "A or C or G or T").Thus these MPR sets will contribute to states and frequencies listedby Trace All States.</P><P>&nbsp;</P><H3><A NAME=14></A>14. Basic Tree and Character Statistics</H3><P>In the formula for treelength (p. 265), <I>n</I> is the number ofcharacters.</P><P>&nbsp;</P><P>In the formula for character RI (p. 269), the numerator anddenominator are inverted. Thus, the formula should read:</P><P>Character RI = <I> F(Mi - si,Mi - mi)</I></P><P>&nbsp;</P><P>The formula for the tree RI (p. 271) is correct.</P><P>&nbsp;</P><H3><A NAME=15></A>15. Charting Tree and Character Statistics</H3><P>A text file of chart calculations (p. 275) can now be written for<B>State Changes &amp; Stasis charts</B> (pp. 287-293) calculatedover a single tree if minimum-maximum changes are requested.</P><P>&nbsp;</P><P><B>Chart To Weight Set </B>(p. 283) is in the <B>Chart</B> menu,not the <B>&#91;&#91;Sigma&#93;&#93;</B> menu.</P><P>&nbsp;</P><P>If you use <B>Chart To Weight Set</B> (pp. 283-284), then thecharacters in any interval with 0 observed steps or changes in achart showing steps or changes will be assigned a weight equal to themaximum value of the scale chosen. Thus, 0 steps using the W = 1/Sweight function and a weight scale from 1 to 10 will be assigned aweight of 10, not a weight of &#91;&#91;infinity&#93;&#93;.Characters with the next-lowest number of steps or changes will beassigned the same, maximal value.</P><P>&nbsp;</P><P>In the <B>Chart To Weight Set</B> (pp. 283-284) and <B>Chart ToType Set </B>(p. 295), MacClade will beep if you enter a value forthe upper end of the weight scale that is too high (see p. 198 forweights, p. 206 for types).</P><P>&nbsp;</P><P>If you ask to have the <B>States</B> chart (p. 296) displayedusing percentages, then in the bar chart and table the values areshown with respect to the pool of observed states (not includingmissing data and gaps). Thus, if frequencies in the data matrix are20% for each of A, C, G, T, and gaps, then the charts would showfrequencies of A, C, G, and T of 25%. The textual summary calculatespercentages in two ways: (1) as in the bar charts and tables; (2)with respect to the total number of data matrix entries (includingmissing data and gaps). The latter method would calculate percentagesof A, C, G, and T as 20% for the example given.</P><P>&nbsp;</P><P>In the <B>Compare 2 Trees</B> and <B>Compare 2 Tree Files</B>charts (pp. 296-297), the vertical axis options dialog box gives youthe option to have the bars indicate either which characters areshorter in a tree or tree file (as illustrated on pp. 296-297 ofbook), or which are longer.</P><P>&nbsp;</P><H3><A NAME=16></A>16. Continuous Characters</H3><P>The maximum number of continuous characters has increased fromfour to ten. These are now chosen in the <B>Edit Continuous</B> and<B>Trace Continuous</B> dialog boxes using scroll bars, not radiobuttons.</P><P>&nbsp;</P><P>Continuous values can be from 0.0 to 9999.9. Larger values andnegative values are not allowed.</P><P>&nbsp;</P><P>The <B>Edit Continuous</B> dialog box (p. 303) allows you toindicate a range for a single terminal taxon by using "to" or "-", asin "0.13 to 0.25" or "2.31 - 3.44". You can also indicate small orlarge numbers using exponential notation, as in "3.5e3" (for 3500.0)or "2.8e-6" (for 0.0000028). However, in the same entry you cannotuse both negative exponents and "-" to indicate a range. Thus "1.4e-6- 2.8e-6" is not allowed and will not be interpreted correctly.</P><P>&nbsp;</P><P>&nbsp;</P><H3><A NAME=18></A>18. Generating Random Data and Random Trees</H3><P><B>Fill data cells with observed states</B>. If you select a blockof cells in the editor, hold down the Option key, and choose <B>FillWith Observed</B> from the <B>Utilities</B> menu, you will bepresented with a dialog box. If you press OK, MacClade will fill theselected cells of each character with random data. If you choose"with probabilities matching observed frequencies", MacClade willfirst tabulate the relative frequencies of each of the statesobserved in a character in all taxa, and fill the selected cells ofthat character with random data, with the probability of filling acell with a state matching the observed frequency of each state. Forexample, if a given nucleotide site had 30% A, 10% G, and 60% T, thenthe selected cells would be filled with the states A, C, G, and Twith probability 0.30, 0.00, 0.10, and 0.60. In contrast, if youchoose "with equal probability for each observed state", then theprobabilities of each state will be equal. In the example given, thecells would be filled with states A, C, G, and T with probabilities0.33, 0.00, 0.33, and 0.33 (note that the probability of C is 0 asthere were no observed C's for this site). If a character has onlymissing data or gaps, then the selected cells will be filled withmissing data.</P><P>&nbsp;</P><P><B>Evolve Characters</B> (pp. 320-322). The number of segmentsthat can be assigned to each branch for the <B>Evolve Characters</B>feature cannot be more than 1000. Also, the WARNING on p. 321 of thebook is no longer applicable, as the number of segments assigned toeach branch is now saved for the current tree. Thus, the number ofsegments on each branch should be saved when you to go to the dataeditor and back, when you save and reopen the file, and so on.However, the numbers of segments are saved only for the tree in thetree window (other trees stored in the file are not stored with theirnumbers of segments). However, in special cases in which a treechange is Undone, the evolve segments may revert to all 1. Thus, youshould still check to see that the segments are as you want them justbefore using them. Also, there is now a "save report" check box thatallows you to choose whether or not to save a report of thesimulation (versions 3.0-3.02 always saved a report).</P><P>&nbsp;</P><P>The <B>Random Trees</B> menu item is in the <B>Trees</B> menu, notthe <B>Edit</B> menu as indicated on page 323.</P><P>&nbsp;</P><H3><A NAME=19></A>19. Using MacClade with Molecular Data</H3><P>For protein-coding nucleotide sequence data, the site number willbe displayed in varied colors in the data editor on a color monitor:blue, green, and red for the first, second, and third codon positionsrespectively. At the bottom left of the editor, beside the autotabbuttons, the numbers "1 2 3" will appear in these colors to provide areminder about the correspondence between colors and codonpositions.</P><P>&nbsp;</P><P>If <B>Translate to Protein</B> (p. 327) is used, MacClade willautomatically name the resulting protein sites. Sites translated fromthe first protein-coding block in the file will be named 1.1, 1.2,1.3, and so on. If in the original nucleotide sequence there was morethan one protein-coding region (separated by non-coding regions),then sites in the second translated region will be named 2.2, 2.3,2.4, and so on.</P><P>&nbsp;</P><P>The <B>Genetic Code</B> dialog box has been altered to allow forseveral new pre-defined genetic codes. In versions 3.0-3.04, youselected the two predefined genetic codes (universal andmitochondrial) using buttons in the dialog box; in version 3.05 and3.06, these have been replaced by a pop-up menu, with eight codes.These include universal, universal extended (equivalent to theuniversal of versions 3.0 to 3.04), Drosophila mitochondrial DNA,Drosophila mitochondrial DNA extended, mammalian mitochondrial DNA,mammalian mitochondrial DNA extended (equivalent to the mtDNA ofversions 3.0 to 3.04), and yeast mitochondrial DNA (Osawa et al.,1992). The extended codes are those in which the disjunct amino acidsare split into distinct states (see p. 332 of the book).</P><P>&nbsp;</P><P>You cannot ask MacClade to create a protein-parsimony user-definedtype (p. 333) if the tree window is on the screen. This can be doneonly if the data editor is on the screen.</P><P>&nbsp;</P><P>See also the note regarding Trace All States under Chapter 13.</P><P>&nbsp;</P><H3><A NAME=20></A>20. Recording Your Work: Printing, Graphics, andText Files</H3><P>Of the various options for altering display and printing of trees(pp. 232-233, 245-246, 256-258, 343-352), some can be controlled inthe tree window, some in the <B>Print Tree</B> dialog box. The rulesare:</P><P>&nbsp;</P><P>1. The controls are separate for the taxon name fonts, whetherpatterns, colors, or grays are used in tracing branches, and alloptions in the <B>Trace Labeling</B> dialog box for Trace All Changesexcept whether ambiguous changes are distinguished. Changing one ofthese options in the tree window will not affect values in the<B>Print Tree</B> dialog box.</P><P>&nbsp;</P><P>2. Controls to change other options in the <B>Trace Labeling</B>dialog boxes are present in both windows, and changing values in onewindow affects the values in the other.</P><P>&nbsp;</P><P>3. Controls for showing node numbers, showing evolve segments,adjusting the content of the taxon labels, and the patterns or colorsused in tracing are available only in the tree window, not in the<B>Print Tree</B> dialog box.</P><P>&nbsp;</P><P>4. Some options are relevant to only one of the windows, and onlyhave controls in that window (e.g., branch lengths).</P><P>&nbsp;</P><H3><A NAME=appendix></A>Appendix 1: File Format</H3><P>The NEXUS file format has changed slightly in the last year, andthese changes are reflected in 3.05-3.06. These changes were made inthe course of preparing the formal description of the NEXUS formatfor publication. They primarily affect the genetic code and codonpositions, continuous data, user-defined types and weights with realvalues, italicized taxon names, and character sets. All of the newfeatures of the file format will be enabled if you choose the "useupdated NEXUS elements" in the Nexus Options dialog box. Some of thenew features (genetic code, codons, italicized taxon names) will beused even if you do not check that option. Use of these new featuresmay make your file unreadable by earlier versions of MacClade.</P><P>&nbsp;</P><P>In addition, the rules regarding punctuation have changed. Bydefault, MacClade 3.06 uses the old punctuation rules on reading afile, the new ones on writing a file (these new files are perfectlylegible to MacClade 3.0-3.04). (The new rules dictate that +, `,&lt;, and &gt; are punctuation, which will break a token; the oldrules did not specify these as punctuation.) If you want to forceMacClade to use the new rules when reading a file, you can do this bygoing to the <B>Save Preferences</B> dialog box in the File menu,uncheck "old punctuation", and press the "Save Preferences" button.The new rules are not currently relevant for any program.</P><P>&nbsp;</P><H3><A NAME=references></A>References</H3><P>The following are corrected references:</P><P>&nbsp;</P><P>Lauder, G. V. 1988. Phylogeny and physiology. <I>Evolution</I>,42:1113-1114.</P><P>Tajima, F. 1983. Evolutionary relationships of DNA sequences infinite populations. <I>Genetics</I>, 105:437-460.</P><P>The following are two new references:</P><P>Kumar, S., K. Tamura, and M. Nei. 1993. MEGA: MolecularEvolutionary Genetics Analysis, version 1.0. The Pennsylvania StateUniversity, University Park, PA 16802.</P><P>Osawa, S., T.H. Jukes, K. Watanabe, and A. Muto. 1992. Recentevidence for the evolution of the genetic code. Microbiol. Rev.56:229-264</P><P></P></BODY></HTML>